src/lib389/lib389/plugins.py

```diff
@@ -827,6 +827,46 @@ def _lint_member_attr_indexes(self):
                         report['check'] = f'memberof:attr_indexes'
                         yield report
 
+    def _lint_member_substring_index(self):
+        if self.status():
+            from lib389.backend import Backends
+            backends = Backends(self._instance).list()
+            attrs = self.get_attr_vals_utf8_l("memberofgroupattr")
+            container = self.get_attr_val_utf8_l("nsslapd-plugincontainerscope")
+            for backend in backends:
+                suffix = backend.get_attr_val_utf8_l('nsslapd-suffix')
+                if suffix == "cn=changelog":
+                    # Always skip retro changelog
+                    continue
+                if container is not None:
+                    # Check if this backend is in the scope
+                    if not container.endswith(suffix):
+                        # skip this backend that is not in the scope
+                        continue
+                indexes = backend.get_indexes()
+                membership_attrs = ['member', 'uniquemember']
+                for attr in membership_attrs:
+                    report = copy.deepcopy(DSMOLE0002)
+                    try:
+                        index = indexes.get(attr)
+                        types = index.get_attr_vals_utf8_l("nsIndexType")
+                        valid = True
+                        if "sub" in types:
+                            valid = False
+
+                        if not valid:
+                            report['detail'] = report['detail'].replace('ATTR', attr)
+                            report['detail'] = report['detail'].replace('BACKEND', suffix)
+                            report['fix'] = report['fix'].replace('ATTR', attr)
+                            report['fix'] = report['fix'].replace('BACKEND', suffix)
+                            report['fix'] = report['fix'].replace('YOUR_INSTANCE', self._instance.serverid)
+                            report['items'].append(suffix)
+                            report['items'].append(attr)
+                            report['check'] = f'attr:substring_index'
+                            yield report
+                    except:
+                        pass
```

Comment:
No, what I mean is that we should not handle bare exceptions like that (and in Python, in general).

I think it should follow this logic (it's just an example, the actual exceptions should be something else):

```python
try:
    result = check()
except ValueError:
    handle_value_error()
except TypeError:
    handle_type_error()
```